arrows/core/video_input_image_list.cxx

```diff
@@ -426,4 +389,133 @@ ::metadata_map()
   return std::make_shared<kwiver::vital::simple_metadata_map>(d->m_metadata_map);
 }
 
+
+// ------------------------------------------------------------------
+void
+video_input_image_list::priv
+::read_from_file( std::string const& filename )
+{
+  typedef kwiversys::SystemTools ST;
+
+  // open file and read lines
+  std::ifstream ifs( filename.c_str() );
+  if ( ! ifs )
+  {
+    throw kwiver::vital::invalid_file( filename, "Could not open file" );
+  }
+  if ( ! this->m_image_reader )
+  {
+    throw kwiver::vital::algorithm_configuration_exception( m_parent->type_name(),
+          m_parent->impl_name(), "invalid image_reader." );
+  }
```

Comment:
Since this check of the image reader is done twice, once in each reader type, couldn't it be done once in ```set_configuration()``` ?

Reply:
Done